version: '3.9'

services:
  database:
    image: postgres:14-alpine
    container_name: n8body.dev-bot-db
    ports:
      - "5430:5432"
    environment:
      - PGUSER=${PGUSER}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
    volumes:
      - n8body.dev-bot-db-volume:/var/lib/postgresql/data
      - ./platform/database/init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      - n8body.dev-bot-net
    healthcheck:
      test: ["CMD-SHELL", "pg_isready", "-d", "${POSTGRES_DB}"]
      interval: 10s
      timeout: 5s
      retries: 5

volumes:
  n8body.dev-bot-db-volume:
    name: n8body.dev-bot-db-volume
    driver: local

networks:
  n8body.dev-bot-net:
    name: n8body.dev-bot-net
    driver: bridge
